sdk/storage/Azure.Storage.Blobs/tests/AppendBlobClientOpenWriteTests.cs

```diff
@@ -0,0 +1,135 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Core.TestFramework;
+using Azure.Storage.Blobs.Models;
+using Azure.Storage.Blobs.Specialized;
+using Azure.Storage.Test;
+using Azure.Storage.Test.Shared;
+using NUnit.Framework;
+
+namespace Azure.Storage.Blobs.Tests
+{
+    public class AppendBlobClientOpenWriteTests : BlobBaseClientOpenWriteTests<AppendBlobClient>
+    {
+        public AppendBlobClientOpenWriteTests(bool async, BlobClientOptions.ServiceVersion serviceVersion)
+            : base(async, serviceVersion, null /* RecordedTestMode.Record /* to re-record */)
+        {
+        }
+
+        protected override AppendBlobClient GetResourceClient(BlobContainerClient container, string resourceName = null, BlobClientOptions options = null)
+        {
+            Argument.AssertNotNull(container, nameof(container));
+
+            string blobName = resourceName ?? GetNewResourceName();
+
+            if (options == null)
+            {
+                return container.GetAppendBlobClient(blobName);
+            }
+
+            container = InstrumentClient(new BlobContainerClient(container.Uri, Tenants.GetNewSharedKeyCredentials(), options ?? ClientBuilder.GetOptions()));
```

Comment:
Why does the container need to be newed up here?

Reply:
This line is only hit if `options` is not null. At this point we need to manually new up a pipeline to apply the new options. We can reformat this code to make it easier to understand the branch that is happening but there does need to be a new client somewhere if options are being explicitly provided.